engine/class_modules/sc_shaman.cpp

```diff
@@ -1485,12 +1530,73 @@ struct shaman_spell_t : public shaman_spell_base_t<spell_t>
       p()->buff.earthen_rage->trigger();
     }
 
-    // BfA Elemental talent - Master of the Elements
     if ( affected_by_master_of_the_elements && !background )
     {
       p()->buff.master_of_the_elements->decrement();
     }
 
+    if ( p()->legendary.elemental_equilibrium->ok() )
+    {
+      // these are general rules based on the tooltip, and should
+      // be overridden per-spell when necessary
+      // they also appear to work on some enhancement spells and almost certainly restoration too
+      if ( dbc::is_school( this->school, SCHOOL_FROST ) )
+      {
+        triggers_elemental_equilibrium_nature = true;
+      }
+
+      if ( dbc::is_school( this->school, SCHOOL_NATURE ) )
+      {
+        triggers_elemental_equilibrium_fire = true;
+      }
+
+      if ( dbc::is_school( this->school, SCHOOL_FIRE ) )
+      {
+        triggers_elemental_equilibrium_frost = true;
+      }
+
+      // only harmful spells can proc/consume the buff
+      if ( execute_state->action->harmful )
```

Comment:
We can remove this guard. Turns out Healing Stream Totem procs it despite everything the tooltip suggests.